tests/Unit/Http/Controllers/Admin/UserControllerTest.php

```diff
@@ -0,0 +1,274 @@
+        $this->be($this->userAuthorized);
+        $request = new UserRequest($this->user);
+        $data = $request->all();
+        $this->userRepo->shouldReceive('getEmailOfUser')
+            ->with($request->email)
+            ->once()
+            ->andReturn(false);
+        $this->userRepo->shouldReceive('create')
+            ->once()
+            ->andReturn(true);
+        $view = $this->userController->store($request);
+        $this->assertEquals(route('admin.users.index'), $view->getTarGetUrl());
+    }
+
+    public function test_store_exist_email_authorize()
+    {
+        $this->be($this->userAuthorized);
+        $request = new UserRequest($this->user);
+        $data = $request->all();
+        $this->userRepo->shouldReceive('getEmailOfUser')
+            ->with($request->email)
+            ->once()
+            ->andReturn($this->user);
+        $view = $this->userController->store($request);
+        $this->assertEquals(route('admin.users.index'), $view->getTarGetUrl());
+    }
+
+    public function test_store_unauthorize()
+    {
+        $this->be($this->userUnAuthorized);
+        $this->expectException(HttpException::class);
+        $request = new UserRequest($this->user);
+        $data = $request->all();
+        $view = $this->userController->store($request);
+        $this->assertEquals(route('admin.users.index'), $view->getTarGetUrl());
+    }
+
+    public function test_edit_exist_view_authorize()
+    {
+        $this->be($this->userAuthorized);
+        $this->userRepo->shouldReceive('find')
+            ->with($this->id)
+            ->once()
+            ->andReturn($this->user);
+        $view = $this->userController->edit($this->id);
+        $this->assertEquals('admin.user.edit', $view->getName());
+    }
+
+    public function test_edit_not_exist_view_authorize()
+    {
+        $this->be($this->userAuthorized);
+        $this->userRepo->shouldReceive('find')
+            ->with($this->id)
+            ->once()
+            ->andReturn(false);
+        $view = $this->userController->edit($this->id);
+        $this->assertEquals(route('admin.users.index'), $view->getTarGetUrl());
+    }
+
+    public function test_edit_unauthorize()
+    {
+        $this->be($this->userUnAuthorized);
+        $this->expectException(HttpException::class);
+        $view = $this->userController->edit($this->id);
+        $this->assertEquals('admin.user.edit', $view->getName());
+    }
+
+    public function test_update_success_authorize()
+    {
+        $this->be($this->userAuthorized);
+        $request = new UserRequest($this->user);
+        $data = [
+            'name' => $request->name,
+            'adrress' => $request->address,
+            'phone' => $request->phone,
+        ];
+        $this->userRepo->shouldReceive('find')
+            ->with($this->id)
+            ->once()
+            ->andReturn($this->user);
+        $this->userRepo->shouldReceive('update')
+            ->withAnyArgs($this->id, $data)
```

Comment:
```suggestion
            ->with($this->id, $data)
```
sao phải dùng **withAnyArgs** chi

Reply:
+ withAnyArgs nó giống như mình không viết vậy nên nó sẽ khhông quan tâm biến mình truyền vào có đúng với kiểu dữ liệu mà mình truyền vào hàm không 
+ with nó sẽ xác định đúng kiểu dữ liệu mà mình truyền vào cho phương thức